### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR... is #928 rebased

## Changes

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/quartz/actions/runs/11297121832#summary-31423391608

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/quartz/actions/runs/11297127173#summary-31423404576

### API Breaks
*  org.quartz.JobDetail.isConcurrentExectionDisallowed -> isConcurrentExecutionDisallowed
* org.quartz.core.QuartzScheduler.notifySchedulerListenersSchduled -> notifySchedulerListenersScheduled
* org.quartz.impl.JobDetailImpl.isConcurrentExectionDisallowed -> isConcurrentExecutionDisallowed
* org.quartz.impl.org.quartz.impl.PROP_DATASOURCE_JNDI_PROVDER -> PROP_DATASOURCE_JNDI_PROVIDER
* org.quartz.impl.jdbcjobstore.AttributeRestoringConnectionInvocationHandler. restoreOriginalAtributes -> restoreOriginalAttributes
* org.quartz.impl.jdbcjobstore.JTANonClusteredSemaphore. DEFAULT_TRANSACTION_MANANGER_LOCATION -> DEFAULT_TRANSACTION_MANAGER_LOCATION
* org.quartz.utils.CircularLossyQueue.isEmtpy -> isEmpty

-----------------
## Checklist
- [ ] tested locally
- [ ] updated the docs
- [ ] added appropriate test
- [x] signed-off on the DCO referenced in the CONTRIBUTING link below via `git commit -s` on my commits, and submit this code under terms of the Apache 2.0 license and assign copyright to the Quartz project owners
  (If you're not using command-line, you can use a [browser extension](https://github.com/scottrigby/dco-gh-ui) )
-----------------
In submitting this contribution, I agree to the terms of contributing as referred to here: 
https://github.com/quartz-scheduler/contributing/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: quartz/src/main/java/org/quartz/JobDetail.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public interface JobDetail extends Serializable, Cloneable {
      * @see DisallowConcurrentExecution
      * @return whether the associated Job class carries the {@link DisallowConcurrentExecution} annotation.
      */
-    public boolean isConcurrentExectionDisallowed();
+    public boolean isConcurrentExecutionDisallowed();
```
Comment (jhouserizer) [inline quartz/src/main/java/org/quartz/JobDetail.java:103]: This breaks public API, should probably think a little about this.

**Comment by jhouserizer** [comment]
Thanks.  A lot of this is useful, a lot superfluous, and a bit potentially harmful (public API breakage).  The latter needs some consideration.

**Comment by jsoref** [comment]
I'm happy to split the API breaks into a separate PR if that helps. Since things are grouped by term, splitting things out isn't hard, all I need is a list.

I'm generally _fairly_ responsive.

**Comment by jhouserizer** [comment]
Thanks @jsoref  - i'll let you know.  I'm probably most inclined to be heavy-handed and just go for it.

**Review by jhouserizer — APPROVED** [review]

**Comment by stefan-caks-dt** [comment]
Hello everyone!

Out of curiousity: Should breaking changes, like in this case renaming of public APIs, even if it's just a spelling mistake, not be a major version increase instead of a minor version increase (assuming you follow SemVer)?

**Comment by msevcenko** [comment]
Hello,

just bumped into this - I would definitely appreciate if public API breaking changes, if they are inevitable, would be at least visibly marked in the changelog ,like here https://github.com/quartz-scheduler/quartz/releases/tag/v2.5.0.

You wouldn't notice the BC in the list of 23 closed issues.

Thank you!

**Pull request merged**